test/server/transformation-controller.spec.js

```diff
@@ -0,0 +1,258 @@
+const expect = require( 'chai' ).expect;
+const request = require( 'supertest' );
+const sinon = require( 'sinon' );
+const communicator = require( '../../app/lib/communicator' );
+const accountModel = require( '../../app/models/account-model' );
+const config = require( '../../app/models/config-model' ).server;
+const surveyModel = require( '../../app/models/survey-model' );
+const userModel = require( '../../app/models/user-model' );
+
+/**
+ * @typedef {import('../../app/models/survey-model').SurveyObject} Survey
+ */
+
+describe( 'Transformation Controller', () => {
+    const basePath = '';
+    const bearer = 'fozzie';
+    const credentials = { bearer };
+    const enketoId = 'surveyZ';
+    const manifestHost = 'http://example.com';
+    const manifestPath = '/manifest';
+    const manifestUrl = `${manifestHost}${manifestPath}`;
+
+    /** @type {import('sinon').SinonSandbox} */
+    let sandbox;
+
+    /** @type {import('express').Application} */
+    let app;
+
+    /** @type {import('http').Server} */
+    let server;
+
+    /** @type {import('../../app/models/account-model').AccountObj */
+    let account;
+
+    /** @type {Survey} */
+    let survey;
+
+    /** @type {Survey} */
+    let surveyWithAccount;
+
+    beforeEach( done => {
+        sandbox = sinon.createSandbox();
+
+        sandbox.stub( config, 'base path' ).get( () => basePath );
```

Comment:
Unless I'm mistaken about some way it could be parameterized in the API, I think that's correct. (And I hope it is, I'm not sure it would be great if that could be parameterized.)